modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java

```diff
@@ -282,25 +253,10 @@ public DistributionZoneManager(
         );
     }
 
-    /** {@inheritDoc} */
     @Override
     public void start() {
-        if (!busyLock.enterBusy()) {
-            throw new IgniteException(NODE_STOPPING_ERR, new NodeStoppingException());
-        }
-
-        try {
-            ZonesConfigurationListener zonesConfigurationListener = new ZonesConfigurationListener();
-
-            zonesConfiguration.distributionZones().listenElements(zonesConfigurationListener);
-            zonesConfiguration.distributionZones().any().dataNodesAutoAdjustScaleUp().listen(onUpdateScaleUp());
-            zonesConfiguration.distributionZones().any().dataNodesAutoAdjustScaleDown().listen(onUpdateScaleDown());
-            zonesConfiguration.distributionZones().any().filter().listen(onUpdateFilter());
-
-            zonesConfiguration.defaultDistributionZone().listen(zonesConfigurationListener);
-            zonesConfiguration.defaultDistributionZone().dataNodesAutoAdjustScaleUp().listen(onUpdateScaleUp());
-            zonesConfiguration.defaultDistributionZone().dataNodesAutoAdjustScaleDown().listen(onUpdateScaleDown());
-            zonesConfiguration.defaultDistributionZone().filter().listen(onUpdateFilter());
+        inBusyLock(busyLock, () -> {
+            registerCatalogEventListenersOnStartManagerBusy();
 
             rebalanceEngine.start();
```

Comment:
I would say that we should start rebalanceEngine at the very end of DZM.start, or more formally after DZM local state recovery.

Reply:
fix it